### ISSUE TITLE
Auto-provision a _deployer account for every new space

### ISSUE THREAD

**Issue by mogul** [18F/cg-product#236]
In order to ensure everyone using cloud.gov has a properly-scoped credential suitable for use in CI/CD systems with little overhead on admins, we want every new space to cause the creation of a _deployer account and communication of the credentials to appropriate people.

**Acceptance Criteria**

- [ ] Scenario: A properly scoped _deployer account is created whenever a new space appears in the system, and its credentials are securely conveyed to users who have the OrgManager role on the enclosing org.  
  - GIVEN a new space has appeared in cloud.gov since the last time the deployer bot ran  
AND the space is not in a sandbox org  
WHEN the deployer bot runs  
THEN a bot auto-provisions a ORGNAME_SPACENAME_deployer account with a strong random password  
AND the bot grants the SpaceDeveloper role to the _deployer account for that space  
AND the bot stashes the _deployer account's password in fugacious.18f.gov  
AND the bot notifies the OrgManagers for the space's org of its existence  
AND the bot informs OrgManagers about compliance concerns about the use of the account (with reference to durable page on this topic in docs.cloud.gov)  
AND the bot informs OrgManagers how to revoke the credentials  
AND the bot informs OrgManagers about how to retrieve the credentials from fugacious  
AND the cloud.gov operators are notified that this happened

**Comment by mogul** [comment]
News flash! It looks like UAA 3.7.0 will include an API for creating user-based API tokens. If Pivotal Tracker is to be believed, it’s due to drop in the next two weeks: https://www.pivotaltracker.com/story/show/128319711

Once that happens, and Travis/Circle/Wercker/Concourse et al support using API tokens for deployment, it may be that we can just send OrgManagers an API token, rather than the credentials for the _deployer account. In that case we may just want to hide or even skip creation of the _deployer user, and instead just provide a button on the dashboard, eg "get an API token for use in deployment to this space".

So: Review and discuss the state of the deployment world before starting work on this story!!

**Comment by afeld** [comment]
Are there any updates on this? Wondering what we should be doing in GovCloud.

**Comment by mogul** [comment]
Nothing was done on this in the Atlas camp since my comment above, probably needs the thread picked up on implementing deployer tokens vs deployer accounts.

**Comment by mogul** [comment]
Looks like [UAA version 3.8.0](https://github.com/cloudfoundry/uaa/releases/tag/3.8.0) just dropped recently. I'm not smart enough to understand whether the token support that made it into there is usable for providing limited scope tokens on the CF side, though.

**Comment by mogul** [comment]
After discussion, superseded by https://github.com/18F/cg-product/issues/437
